### ISSUE TITLE
False Positive E1102: time.time is not callable (not-callable)

### ISSUE THREAD

**Issue by jamesquilty** [PyCQA/pylint#10874]
### Bug description

I have a module in my source tree `src/util/time.py` which defines some utility functions

```python
import time

def somefunc():
    ...
```

in `src/main.py` I have

```python
import time
from util.time import somefunc
...
def main():
    ...
    current_time = time.time()
```

and pylint is complaining about every occurrence of `time.time()`.

I'm not entirely happy with this code and I'd like to refactor `src/util/time.py` to

```python
from time import *

def somefunc():
    ...
```

which avoids pylint throwing spurious E1102 errors, but it seems to me that pylint shouldn't be emitting E1102 in this circumstance in the first place.

### Configuration

```ini

```

### Command used

```shell
pylint src
```

### Pylint output

```python
************* Module main
src/main.py:368:59: E1102: time.time is not callable (not-callable)
src/main.py:453:34: E1102: time.time is not callable (not-callable)
src/main.py:499:19: E1102: time.time is not callable (not-callable)
src/main.py:508:34: E1102: time.time is not callable (not-callable)
```

### Expected behaviour

The error is not emitted.

### Pylint version

```shell
pylint 4.0.5
astroid 4.0.4
Python 3.11.9 (main, May 11 2024, 12:27:00) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

macOS 15.7.4

### Additional dependencies

```python

```

**Comment by jacobtylerwalls** [comment]
Duplicate of #6535
